### PR TITLE
feat(nxls): allow nxls to kill itself after 3h of inactivity to prevent background resource usage

### PR DESCRIPTION
**Pull request by MaxKless**
- feat(nxls): allow nxls to kill itself after 3h of inactivity to prevent background resource usage
- flag behaviour behind initializationOption

### CODE REVIEW

File: apps/vscode/src/refresh-workspace.ts
Comment (jaysoo) [inline apps/vscode/src/refresh-workspace.ts:21]: The try-catch-finally isn't used since both `.refreshWorkspace()` and `.restart()` has the try-catch. It's possible in the future that logic added to those methods land outside of their respective try-catch blocks, thus causing `isRefreshing = false` to not execute.

**Review by jaysoo — APPROVED** [review]

**Comment by MaxKless** [comment]
STILL NEEDS MORE INVESTIGATION AFTER REBASE - DO NOT MERGE

**Pull request merged**
